src/test/java/org/opensearch/knn/index/codec/KNN990Codec/KNNQuantizationStateReaderTests.java

```diff
@@ -0,0 +1,236 @@
+            mockedStaticReader.when(() -> KNNQuantizationStateReader.readStateBytes(any(IndexInput.class), anyLong(), anyInt()))
+                .thenReturn(new byte[8]);
+            try (MockedStatic<CodecUtil> mockedStaticCodecUtil = mockStatic(CodecUtil.class)) {
+                assertThrows(IllegalArgumentException.class, () -> KNNQuantizationStateReader.read(quantizationStateReadConfig));
+
+                mockedStaticCodecUtil.verify(() -> CodecUtil.retrieveChecksum(input));
+                Mockito.verify(input, times(4)).readInt();
+                Mockito.verify(input, times(2)).readVLong();
+
+                Mockito.when(input.readInt()).thenReturn(fieldNumber);
+
+                try (MockedStatic<OneBitScalarQuantizationState> mockedStaticOneBit = mockStatic(OneBitScalarQuantizationState.class)) {
+                    Mockito.when(quantizationStateReadConfig.getQuantizationParams()).thenReturn(scalarQuantizationParams1);
+                    OneBitScalarQuantizationState oneBitScalarQuantizationState = Mockito.mock(OneBitScalarQuantizationState.class);
+                    mockedStaticOneBit.when(() -> OneBitScalarQuantizationState.fromByteArray(any(byte[].class)))
+                        .thenReturn(oneBitScalarQuantizationState);
+                    QuantizationState quantizationState = KNNQuantizationStateReader.read(quantizationStateReadConfig);
+                    assertTrue(quantizationState instanceof OneBitScalarQuantizationState);
+                }
+
+                try (MockedStatic<MultiBitScalarQuantizationState> mockedStaticOneBit = mockStatic(MultiBitScalarQuantizationState.class)) {
+                    MultiBitScalarQuantizationState multiBitScalarQuantizationState = Mockito.mock(MultiBitScalarQuantizationState.class);
+                    mockedStaticOneBit.when(() -> MultiBitScalarQuantizationState.fromByteArray(any(byte[].class)))
+                        .thenReturn(multiBitScalarQuantizationState);
+
+                    Mockito.when(quantizationStateReadConfig.getQuantizationParams()).thenReturn(scalarQuantizationParams2);
+                    Mockito.when(quantizationStateReadConfig.getQuantizationParams()).thenReturn(scalarQuantizationParams2);
+                    QuantizationState quantizationState = KNNQuantizationStateReader.read(quantizationStateReadConfig);
+                    assertTrue(quantizationState instanceof MultiBitScalarQuantizationState);
+
+                    Mockito.when(quantizationStateReadConfig.getQuantizationParams()).thenReturn(scalarQuantizationParams4);
+                    Mockito.when(quantizationStateReadConfig.getQuantizationParams()).thenReturn(scalarQuantizationParams4);
+                    quantizationState = KNNQuantizationStateReader.read(quantizationStateReadConfig);
+                    assertTrue(quantizationState instanceof MultiBitScalarQuantizationState);
```

Comment:
instead of instanceof, can we do `assertEquals(expectedState, quantizationState)`?